### PR TITLE
Remove external link interstitial functionality 🎉 

### PR DESCRIPTION
**Pull request by anselmbradford**
Fixes https://github.com/cfpb/consumerfinance.gov/issues/7845

## Removals

- Removes external link interstitial functionality 🎉 

## How to test this PR

1. PR checks should pass.
2. Any external link should get an external link icon, but not have an interstitial.

### CODE REVIEW

File: cfgov/core/utils.py [hunk 1]
```diff
@@ -65,34 +62,12 @@
...
-    elif match.group("domain") in settings.ALLOWED_LINKS_WITHOUT_INTERSTITIAL:
-        return False
-
-    return True
-
-
 def sign_url(url):
     signer = Signer(sep="||")
     url, signature = signer.sign(url).split("||")
     return (url, signature)
```
Comment (willbarton) [inline cfgov/core/utils.py:68]: Without the interstitial, `sign_url` should also be unnecessary.

**Review by willbarton — COMMENTED** [review]
One minor thing that can also go, but I think this looks good!

I think I had repressed or blocked out any knowledge of `data-pretty-href`, and it makes me 😱 .

**Review by chosak — APPROVED** [review]
🥇

+1 to Will's suggestion, but besides, that, this looks great! PR of the year so far!

File: cfgov/core/utils.py (hunk 1 above)
Comment (anselmbradford) [reply cfgov/core/utils.py]: https://github.com/cfpb/consumerfinance.gov/pull/8095/commits/b7de2d452e4793722331e7d9af738f95bafd431d

**Review by willbarton — APPROVED** [review]

**Pull request merged**